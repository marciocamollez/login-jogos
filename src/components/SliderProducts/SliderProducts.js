import React from "react";
import { Link } from "react-router-dom";
import Carousel from "react-multi-carousel";
import "react-multi-carousel/lib/styles.css";
import styles from "./SliderProducts.module.scss";

function SliderProducts({ games }) {
  const responsive = {
    superLargeDesktop: {
      breakpoint: { max: 4000, min: 3000 },
      items: 5,
    },
    desktop: {
      breakpoint: { max: 3000, min: 1024 },
      items: 3,
    },
    tablet: {
      breakpoint: { max: 1024, min: 464 },
      items: 2,
    },
    mobile: {
      breakpoint: { max: 464, min: 0 },
      items: 1,
    },
  };

  return (
    <div>
      <Carousel
        responsive={responsive}
        autoPlay={true}
        autoPlaySpeed={2000}
        infinite={true}
      >
        {games.map((game) => (
          <section key={game.id} className={styles.slider}>
            <div>
              <Link to={`/game/${game.id}`}>
                <img src={game.img} alt={game.nome} title={game.nome} />
              </Link>
            </div>
          </section>
        ))}
      </Carousel>
    </div>
  );
}

export default SliderProducts;
