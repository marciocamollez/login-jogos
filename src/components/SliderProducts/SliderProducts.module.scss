@import "../../styles/Variables.scss";

.slider{
    display: flex;
    padding: 0 20px;
    box-sizing: border-box;

    div{
        img{
            max-width: 100%;
            display: block;
            border-radius: 6px;
        }
    }
}