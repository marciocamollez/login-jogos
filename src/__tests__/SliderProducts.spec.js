import { render, screen } from "@testing-library/react";
import SliderProducts from "../components/SliderProducts/SliderProducts";
import { MemoryRouter } from "react-router-dom";

describe("SliderProducts Component", () => {
  test("should render slider products div", () => {
    render(<SliderProducts />, { wrapper: MemoryRouter });
    const el = screen.getByTestId("test-slider");
    expect(el).toBeInTheDocument();
    expect(el).toMatchSnapshot();
    expect(el).toBeTruthy();
  });
});
