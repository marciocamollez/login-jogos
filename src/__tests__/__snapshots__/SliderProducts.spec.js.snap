// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SliderProducts Component should render slider products div 1`] = `
<div
  data-testid="test-slider"
>
  <div
    class="react-multi-carousel-list  "
    dir="ltr"
  >
    <ul
      class="react-multi-carousel-track "
      style="transition: none; overflow: unset; transform: translate3d(0px,0,0);"
    />
  </div>
</div>
`;
